core/src/test/scala/unit/kafka/cluster/PartitionTest.scala

```diff
@@ -1149,8 +1149,8 @@ class PartitionTest extends AbstractPartitionTest {
 
     // let the follower in ISR move leader's HW to move further but below LEO
     fetchFollower(partition, replicaId = follower2, fetchOffset = 0)
-    fetchFollower(partition, replicaId = follower2, fetchOffset = lastOffsetOfFirstBatch)
-    assertEquals(lastOffsetOfFirstBatch, partition.log.get.highWatermark, "Expected leader's HW")
+    fetchFollower(partition, replicaId = follower2, fetchOffset = lastOffsetOfFirstBatch + 1)
```

Comment:
Similar thing happening here as the UnifiedLogTest `testOffsetSnapshot`. `lastOffsetOfFirstBatch == 1` which is not the first record in a batch, so this was returning 0, which is the first record in batch 1.